packages/admin/admin/src/form/FinalFormChooseFile.tsx

```diff
@@ -0,0 +1,266 @@
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        droppableAreaIsDisabled: {},
+        droppableAreaHasError: {},
+        fileListText: {
+            textOverflow: "ellipsis",
+            overflow: "hidden",
+            whiteSpace: "nowrap",
+        },
+    });
+};
+
+export interface FinalFormChooseFileProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+    disableDropzone?: boolean;
+    disableSelectFileButton?: boolean;
+    accept: Accept;
+    maxSize: number;
+    maxFiles?: number;
+    iconMapping?: {
+        delete?: React.ReactNode;
+        info?: React.ReactNode;
+        select?: React.ReactNode;
+    };
+}
+
+const FinalFormChooseFileComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormChooseFileProps> = ({
+    classes,
+    disabled,
+    disableDropzone,
+    disableSelectFileButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    maxFiles,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
+}) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Info color="error" />, select: selectIcon = <Select /> } = iconMapping;
+
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            if (multipleFiles && Array.isArray(fieldValue)) {
+                if (!maxFiles || (maxFiles && fieldValue.length < maxFiles && fieldValue.length + acceptedFiles.length <= maxFiles)) {
+                    onChange([...fieldValue, ...acceptedFiles]);
+                } else onChange([...acceptedFiles]);
+            } else onChange([...acceptedFiles]);
+        },
+        [fieldValue, multipleFiles, onChange, maxFiles],
+    );
+
+    const removeFile = (removedFile: File) => () => {
+        const newFiles = Array.isArray(fieldValue) ? fieldValue.filter((file) => file !== removedFile) : undefined;
+        onChange(newFiles);
+    };
+
+    const { fileRejections, getRootProps, getInputProps, isDragReject } = useDropzone({
+        onDrop,
+        accept,
+        disabled: disabled || (maxFiles && fieldValue.length >= maxFiles),
+        multiple: multipleFiles,
+        maxSize: maxSize,
+        maxFiles,
+    });
+
+    let acceptedFiles: File[] = [];
+
+    if (Array.isArray(fieldValue)) {
+        acceptedFiles = fieldValue;
+    } else if (fieldValue.name !== undefined) {
+        acceptedFiles = [fieldValue];
+    }
+
+    const rejectedFiles = fileRejections.map((rejectedFile) => (
+        <div key={rejectedFile.file.name} className={classes.rejectedFileListItem}>
+            <div className={classes.fileListText}>{rejectedFile.file.name}</div>
+            <IconButton color="error">{infoIcon}</IconButton>
```

Comment:
This button is missing `onClick`. Is that missing or should this just be an icon?